simple-page-ordering.php

```diff
@@ -15,676 +15,19 @@
  * @package simple-page-ordering
  */
 
-namespace SimplePageOrdering;
-
-use stdClass;
-use WP_Error;
-use WP_REST_Response;
-use WP_Query;
-
-/**
- * Get the minimum version of PHP required by this plugin.
- *
- * @since 2.5.2
- *
- * @return string Minimum version required.
- */
-function minimum_php_requirement(): string {
-	return '7.4';
-}
-
-/**
- * Whether PHP installation meets the minimum requirements
- *
- * @since 2.5.2
- *
- * @return bool True if meets minimum requirements, false otherwise.
- */
-function site_meets_php_requirements(): bool {
-	return version_compare( phpversion(), minimum_php_requirement(), '>=' );
-}
-
-// Try to load the plugin files, ensuring our PHP version is met first.
-if ( ! site_meets_php_requirements() ) {
-	add_action(
-		'admin_notices',
-		function() {
-			?>
-			<div class="notice notice-error">
-				<p>
-					<?php
-					echo wp_kses_post(
-						sprintf(
-						/* translators: %s: Minimum required PHP version */
-							__( 'Simple Page Ordering requires PHP version %s or later. Please upgrade PHP or disable the plugin.', 'simple-page-ordering' ),
-							esc_html( minimum_php_requirement() )
-						)
-					);
-					?>
-				</p>
-			</div>
-			<?php
-		}
-	);
+if ( ! is_readable( __DIR__ . '/vendor/autoload.php' ) ) {
```

Comment:
In the event further composer dependencies are added at a later date includingthis here may cause problems if the new dependencies require > PHP 5.6

Would it be possible to do one of the following:
* only require the compatibility checker in the vendor path
* install the compatibility checker outside of `vendor` and require it from the custom location

Reply:
@peterwilsoncc I've taken the approach to move the library from `vendor` to `10up-lib` directory and also loaded the SPO main class after the validation is completed.